### PR TITLE
Nix on SmartOS updates (pre nixconf)

### PR DESCRIPTION
**Pull request by vizanto**
Hope this is not too big to merge :-)

### CODE REVIEW

File: pkgs/build-support/fetchgit/builder.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ source $stdenv/setup
 
 header "exporting $url (rev $rev) into $out"
 
-$fetcher --builder --url "$url" --out "$out" --rev "$rev" \
+$SHELL $fetcher --builder --url "$url" --out "$out" --rev "$rev" \
```
Comment (vizanto) [inline pkgs/build-support/fetchgit/builder.sh]: I'll try if this is still needed

**Comment by domenkozar** [comment]
You should do a PR against staging, as this is a stdenv rebuild

File: pkgs/build-support/fetchgit/builder.sh (hunk 1 above)
Comment (vizanto) [inline pkgs/build-support/fetchgit/builder.sh]: Doesn't seem to trigger a rebuild, so should be more reliable.

**Comment by vizanto** [comment]
Ok, I'll rebase on staging @domenkozar

**Pull request closed without merging**